### ISSUE TITLE
Icon in ComboBox incorrectly aligned

### ISSUE THREAD

**Issue by mlopezFC** [vaadin/framework#9637]
Hi!
There is a visual problem in comboboxes filled with icons (Vaadin Version: 8.1.0.beta2).
The selected item's icon, appears very close to the left edge, as seen in the attached screenshot:

![comboboxiconproblem](https://user-images.githubusercontent.com/28312758/27838286-f4184e14-60bf-11e7-8eb9-f268caa6e715.png)
This is the code to reproduce the issue:

    @SuppressWarnings("serial")
    @Theme("mytheme")
    public class MyUI extends UI {
    	public enum PhoneType {
    		LANDLINE(VaadinIcons.PHONE_LANDLINE), MOBILE(VaadinIcons.MOBILE), FAX(VaadinIcons.FILE_TEXT_O);
    		private VaadinIcons icon;
    		private PhoneType(VaadinIcons icon) {
    			this.icon = icon;
    		}
    		public Resource getIcon() {
    			return icon;
    		}
    	}
    	
    @Override
    protected void init(VaadinRequest vaadinRequest) {
        final VerticalLayout layout = new VerticalLayout();       
		ComboBox<PhoneType> type = new ComboBox<>();
		type.setItems(PhoneType.values());
		type.setEmptySelectionAllowed(false);
		type.setValue(PhoneType.LANDLINE);
		type.setItemIconGenerator(new IconGenerator<PhoneType>() {
			@Override
			public Resource apply(PhoneType item) {
				return item.getIcon();
			}
		});
		
		layout.addComponent(type);
		setContent(layout);
        }
        
        @WebServlet(urlPatterns = "/*", name = "MyUIServlet", asyncSupported = true)
        @VaadinServletConfiguration(ui = MyUI.class, productionMode = false)
        public static class MyUIServlet extends VaadinServlet {
        }
    }

Adding this in styles.scss, seems to fix the problem, but I don't know if it's the correct way to solve it:

    .v-filterselect span.v-icon {
    	padding-left: 0.5em;	
    }

Regards!

**Comment by hesara** [comment]
Indeed, the valo theme applies left padding for img.v-icon (image resource based icons) in a ComboBox, but not for span.v-icon (font icon) - see https://github.com/vaadin/framework/blob/87998f5ae7f31350dc08be150fdff43be4c1f331/themes/src/main/themes/VAADIN/themes/valo/components/_combobox.scss#L150 .

I believe a similar calculated margin-left could and probably should be applied for font icons (span.v-icon).

**Comment by sebastianfilke** [comment]
I think its related to this: #9072
